app/models/spotlight/reindex_progress.rb

```diff
@@ -0,0 +1,101 @@
+module Spotlight
+  ##
+  # ReindexProgress is a class that models the progress of reindexing a list of resources
+  class ReindexProgress
+    MINUTES_IN_PROGRESS = 10
+
+    def initialize(resource_list)
+      @resources = if resource_list.present?
+                     resource_list
+                   else
+                     null_resources
+                   end
+    end
+
+    def in_progress?
+      return unless finished
+      finished > MINUTES_IN_PROGRESS.minutes.ago
```

Comment:
This basically will give us a window of 10 minutes after the last item was finished indexing where the panel will be present.  Not sure if we should use a different amount of time or a different algorithm all together.